kernel/mem_domain.c

```diff
@@ -13,6 +13,19 @@
 static u8_t max_partitions;
 
 
+static void ensure_w_xor_x(u32_t attrs) {
```

Comment:
in Zephyr it's customary to do little or no runtime checking.
Seems to me that it might be better to wrap this in an __ASSERT()

Reply:
Changed to an assert.